src/www/ui/api/Controllers/UploadController.php

```diff
@@ -810,6 +811,10 @@ public function getGroupsWithPermissions($request, $response, $args)
     $finalPermGroups = [];
     foreach ($permGroups as $value) {
       unset($value["perm_upload_pk"]);
+      $value["groupPk"] = $value["group_pk"];
+      unset($value["group_pk"]);
+      $value["groupName"] = $value["group_name"];
+      unset($value["group_name"]);
       $finalPermGroups[] = $value;
```

Comment:
I would instead recommend to create a new model and get the version aware array from it. See https://github.com/fossology/fossology/pull/2617/files#diff-f62f1d2b860c504c87d386cdfa79513c109848593f0afeb60a5dc76e4b98ff42 for reference.